### ISSUE TITLE
Disconnect Hangs After Using Transactions

### ISSUE THREAD

**Issue by dan-gittik** [encode/databases#262]
This program hangs for me on the `disconnect` call (until I hit CTRL+C; tracebacks are attached later).

```python
import asyncio
import databases

async def main():
    db = Database(url)
    await db.connect()
    transaction = await db.transaction()
    await transaction.start()
    print(await db.execute('select 1'))
    await transaction.commit() # or rollback, doesn't matter
    await db.disconnect()

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

The same program works if I remove the explicit transaction:

```python
import asyncio
import databases

async def main():
    db = Database(url)
    await db.connect()
    print(await db.execute('select 1'))
    await db.disconnect()

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

And it also works if I use a context manager:

```python
import asyncio
import databases

async def main():
    db = Database(url)
    await db.connect()
    async with db.transaction():
        print(await db.execute('select 1'))
    await db.disconnect()

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

I'm using Python 3.8.1 on Mac OS X 10.14.6, and for the URL I used `sqlite:////tmp/db.sqlite3` and `postgrseql://user:1234@localhost:5432/test` (on a standard postgres Docker container). These were the tracebacks after I hit CTRL+C on the hanging program:

```
# for sqlite3
Exception ignored in: <module 'threading' from '/Users/dan/.pyenv/versions/3.8.1/lib/python3.8/threading.py'>
Traceback (most recent call last):
  File "/Users/dan/.pyenv/versions/3.8.1/lib/python3.8/threading.py", line 1388, in _shutdown
    lock.acquire()
KeyboardInterrupt:

# for postgresql
Traceback (most recent call last):
  File "a.py", line 19, in <module>
    loop.run_until_complete(main())
  File "/Users/dan/.pyenv/versions/3.8.1/lib/python3.8/asyncio/base_events.py", line 599, in run_until_complete
    self.run_forever()
  File "/Users/dan/.pyenv/versions/3.8.1/lib/python3.8/asyncio/base_events.py", line 567, in run_forever
    self._run_once()
  File "/Users/dan/.pyenv/versions/3.8.1/lib/python3.8/asyncio/base_events.py", line 1819, in _run_once
    event_list = self._selector.select(timeout)
  File "/Users/dan/.pyenv/versions/3.8.1/lib/python3.8/selectors.py", line 558, in select
    kev_list = self._selector.control(None, max_ev, timeout)
KeyboardInterrupt
```

After fiddling a bit with the source code (specifically, I was trying to figure out why the context manager works, but the methods themselves don't), it seems like the problem might be with the documentation — [here](https://www.encode.io/databases/connections_and_transactions/) it says to `await` the `db.transaction()` as well as its `start()`, but that seems to cause the problem. If I drop the first `await`, this works:

```python
import asyncio
import databases

async def main():
    db = Database(url)
    await db.connect()
    transaction = db.transaction()
    await transaction.start()
    print(await db.execute('select 1'))
    await transaction.commit() # or rollback, doesn't matter...
    await db.disconnect()

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

So maybe everything's working as intended, but then this tiny typo in the documentation should probably be fixed.
Beautiful library, by the way, thanks for making it happen :)

**Comment by acnebs** [comment]
I have noticed the same thing, though I actually was still not having success even with the adjustment used by @dan-gittik, as I was trying to use the ____ pattern.

I thought I was having success with a pattern like this:

```python
    transaction = await pg.transaction().start()
    try:
        yield pg
    except Exception as e:
        await transaction.rollback()
        deal_with_exception(e)
    else:
        await transaction.commit()
```

But on retrospect it wasn't actually creating a transaction (rollbacks were not happening on exception). 
Maybe I'm missing something, but I don't really understand how this code makes sense:

```python
    def __await__(self) -> typing.Generator:
        """
        Called if using the low-level `transaction = await database.transaction()`
        """
        return self.start().__await__()
```

The `start()` method returns a `Transaction`, and `__await__` just calls `start().__await__()` -> is this not entirely circular logic? Maybe I'm just not well versed enough in co-routines.

**Comment by arauter** [comment]
@dan-gittik your recommendation to use:
```
transaction = db.transaction()
```

was a life safer.

**Comment by goteguru** [comment]
In the source we can see Database.transaction() is not a coroutine,  awaiting for it is unreasonable. It's just a thin wrapper returning the Transaction object for the current connection which is also an async context manager. 
```
await db.transaction().start()
```
or
```
t = db.transaction()
await t.start()
```
or 
```
async with db.transaction():
    ....
```
are equally correct. `await db.transaction()` is not. 
(Should be fixed in the docs)

**Comment by aminalaee** [comment]
Fixed by #390
